tapgarden/seedling.go

```diff
@@ -101,7 +101,8 @@ type Seedling struct {
 // validateFields attempts to validate the set of input fields for the passed
 // seedling, an error is returned if any of the fields are out of spec.
 //
-// TODO(roasbeef): have this series of check be a DB level constraint?
+// NOTE: This function does not check the group key. That check is performed in
```

Comment:
I think what the TODO actually meant was to add a `CHECK(length(asset_name) > 0)` and `CHECK(asset_supply > 0)` and `CHECK(asset_type = 0 OR asset_type = 1)` to the DB. But IMO we shouldn't have too much logic on the DB level, so I'm fine with this change.

Reply:
I saw that we have `NOT NULL` which should cover `CHECK(length(asset_name) > 0)` and `CHECK(asset_supply > 0)`.

Would you be opposed to me adding `CHECK(asset_type = 0 OR asset_type = 1)`?

Reply:
No, in SQL an empty string is not the same as `NULL`. Neither is the value 0. So there is a diff here.

Reply:
Good point.

I'm keen on SQL level checks myself, because they can be reused in a different implementation of Taproot Assets. But that probably won't happen for a while!